packages/apps/desktop-viewer-test/src/frontend/components/routes/ViewerRoute.tsx

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Viewer } from "@itwin/desktop-viewer-react";
+import { RouteComponentProps } from "@reach/router";
+import React, { useEffect, useMemo, useState } from "react";
+
+import { IModelMergeItemsProvider } from "../../extensions";
+
+interface ViewerRouteProps extends RouteComponentProps {
+  children?: any;
+}
+
+export interface ViewerRouteState {
+  filePath?: string;
+}
+
+export const ViewerRoute = ({ location }: ViewerRouteProps) => {
+  const [filePath, setFilePath] = useState<string>();
+
+  useEffect(() => {
+    const routeState = location?.state as ViewerRouteState | undefined;
+    if (routeState?.filePath) {
+      setFilePath(routeState?.filePath);
+    }
+  }, [location?.state]);
+
+  const onIModelAppInitialized = useMemo(() => {
+    console.log("iTwin.js Initialized!");
+  }, []);
+
+  return filePath ? (
+    <Viewer
+      snapshotPath={filePath}
+      onIModelAppInit={onIModelAppInitialized as any}
```

Comment:
did you need to cast it as any? was tsc complaining?

Reply:
its because it's memoized...but we should really require it to be. Maybe we should change the type accordingly. I'll look into it

Reply:
Thinking about it, should probably use useCallback instead of useMemo

Reply:
yeah you're right.....I dont know why useMemo returns void, but I guess thats to force usecallback for functions

Reply:
fixed